online-shop/src/App.tsx

```diff
@@ -1,31 +1,58 @@
 import React from 'react';
-import './Styles/Chapter3Style.css';
-import ProductDetails from './Components/ProductDetails';
-import ProductList from './Components/ProductList';
-import './Styles/ComponentsStyles/App.sass';
+import './Styles/ComponentsStyles/AppStyles/App.sass';
 import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
-import ShoppingCart from './Components/ShoppingCart';
-import EditProductPage from './Components/EditProductPage';
 import Navbar from './HelperComponents/Navbar';
 import './App.scss';
-import NewProductInitializer from './Components/NewProduct';
+import SalesViewInitializer from './Components/SalesComponent/SalesSmartView';
+import ProductListViewInitializer from './Components/ProductListComponent/ProductListSmartView';
+import ProductDetailsViewInitializer from './Components/ProductDetailsComponent/ProductDetailsSmartView';
+import ShoppingCartViewInitializer from './Components/ShoppingCartComponent/ShoppingCartSmartView';
+import EditProductPageViewInitializer from './Components/EditProductPageComponent/EditProductSmartView';
+import NewProductViewInitializer from './Components/NewProductComponent/NewProductSmartView';
+import { AppState } from './ReduxStore';
+import { connect } from 'react-redux';
+import { LoginComponentDumpView } from './Components/LoginComponent/LoginComponentDumpView';
+import LoginComponentViewInitializer from './Components/LoginComponent/LoginComponentSmartView';
+
+interface AppComponentState {
+  isLoggedIn: boolean;
+}
+
+class App extends React.Component<AppComponentState> {
+  checkIsLoggedIn(): boolean {
+    //username doej, password: password
+    console.log("Is someone logged in? " + this.props.isLoggedIn);
+    return this.props.isLoggedIn;
+  }
 
-export default class App extends React.Component {
   render() {
     return (
-      <Router>
+      <Router key="General Router">
         <div className="App">
-          <Navbar />
+          <Navbar isHidden={this.props.isLoggedIn} />
           <Switch>
             <Redirect exact from='/' to='/products' />
-            <Route path="/products" exact render = {(props) => <ProductList {...props} />}/>
-            <Route path="/products/:id" exact render={(props) => <ProductDetails {...props} />} />
-            <Route path="/shoppingCart" exact render={(props) => <ShoppingCart match = {props.match}/>} />
-            <Route path="/editProduct/:id" exact render={(props) => <EditProductPage match = {props.match}/>} />
-            <Route path="/newProduct" exact render={(props) => <NewProductInitializer match = {props.match}/>}/>
+            <Route path="/products" exact component={this.checkIsLoggedIn() ? ProductListViewInitializer : LoginComponentViewInitializer} />
```

Comment:
Instead of checking if the user is logged in on each route, you could simply use an if statement and just render the login route if the user is not logged in.